google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/changestream/ChangeStreamStateMachine.java

```diff
@@ -0,0 +1,580 @@
+                setCell.getColumnQualifier(),
+                setCell.getTimestampMicros());
+            return AWAITING_CELL_VALUE.handleMod(dataChange, index);
+          }
+          // Case 2: DeleteFamily
+          if (mod.hasDeleteFromFamily()) {
+            numNonCellMods++;
+            builder.deleteFamily(mod.getDeleteFromFamily().getFamilyName());
+            return checkAndFinishMutationIfNeeded(dataChange, index + 1);
+          }
+          // Case 3: DeleteCell
+          if (mod.hasDeleteFromColumn()) {
+            numNonCellMods++;
+            builder.deleteCells(
+                mod.getDeleteFromColumn().getFamilyName(),
+                mod.getDeleteFromColumn().getColumnQualifier(),
+                TimestampRange.create(
+                    mod.getDeleteFromColumn().getTimeRange().getStartTimestampMicros(),
+                    mod.getDeleteFromColumn().getTimeRange().getEndTimestampMicros()));
+            return checkAndFinishMutationIfNeeded(dataChange, index + 1);
+          }
+          throw new IllegalStateException("AWAITING_NEW_MOD: Unexpected mod type");
+        }
+      };
+
+  /**
+   * A state that represents a cell's value continuation.
+   *
+   * <dl>
+   *   <dt>Valid exit states:
+   *   <dd>{@link ChangeStreamStateMachine#AWAITING_NEW_MOD}. Current chunked SetCell is added, and
+   *       we have more mods to expect.
+   *   <dd>{@link ChangeStreamStateMachine#AWAITING_CELL_VALUE}. Current chunked SetCell has more
+   *       cell values to expect.
+   *   <dd>{@link ChangeStreamStateMachine#AWAITING_STREAM_RECORD_CONSUME}. Current chunked SetCell
+   *       is the last mod of the current logical mutation.
+   * </dl>
+   */
+  private final State AWAITING_CELL_VALUE =
+      new State() {
+        @Override
+        State handleHeartbeat(ReadChangeStreamResponse.Heartbeat heartbeat) {
+          throw new IllegalStateException(
+              "AWAITING_CELL_VALUE: Can't handle a Heartbeat in the middle of building a SetCell.");
+        }
+
+        @Override
+        State handleCloseStream(ReadChangeStreamResponse.CloseStream closeStream) {
+          throw new IllegalStateException(
+              "AWAITING_CELL_VALUE: Can't handle a CloseStream in the middle of building a SetCell.");
+        }
+
+        @Override
+        State handleMod(ReadChangeStreamResponse.DataChange dataChange, int index) {
+          validate(
+              0 <= index && index <= dataChange.getChunksCount() - 1,
+              "AWAITING_CELL_VALUE: Index out of bound.");
+          ReadChangeStreamResponse.MutationChunk chunk = dataChange.getChunks(index);
+          validate(
+              chunk.getMutation().hasSetCell(),
+              "AWAITING_CELL_VALUE: Current mod is not a SetCell.");
+          Mutation.SetCell setCell = chunk.getMutation().getSetCell();
+          numCellChunks++;
+          builder.cellValue(setCell.getValue());
+          // Case 1: Current SetCell is chunked.
```

Comment:
For example: [ReadChangeStreamResponse1: {DC, DF, SC1}, ReadChangeStreamResponse2: {SC2, DF}]

Reply:
Done.